app/code/Magento/Store/Test/Unit/Controller/Store/RedirectTest.php

```diff
@@ -0,0 +1,280 @@
+    {
+        $this->requestMock
+            ->expects($this->at(0))
+            ->method('getParam')
+            ->with(StoreResolver::PARAM_NAME)
+            ->willReturn($storeCode);
+        $this->requestMock
+            ->expects($this->at(1))
+            ->method('getParam')
+            ->with('___from_store')
+            ->willReturn($defaultStoreViewCode);
+        $this->requestMock
+            ->expects($this->at(2))
+            ->method('getParam')
+            ->with(ActionInterface::PARAM_NAME_URL_ENCODED)
+            ->willReturn($defaultStoreViewCode);
+        $this->storeRepositoryMock
+            ->expects($this->once())
+            ->method('get')
+            ->with($defaultStoreViewCode)
+            ->willReturn($this->formStoreMock);
+        $this->formStoreMock
+            ->expects($this->once())
+            ->method('getCode')
+            ->willReturnSelf();
+        $this->sidResolverMock
+            ->expects($this->once())
+            ->method('getUseSessionInUrl')
+            ->willReturn(false);
+        $this->hashGeneratorMock
+            ->expects($this->once())
+            ->method('generateHash')
+            ->with($this->formStoreMock)
+            ->willReturn([]);
+
+        $this->redirectMock
+            ->expects($this->once())
+            ->method('redirect')
+            ->with(
+                $this->responseMock,
+                'stores/store/switch',
+                ['_nosid' => true,
+                    '_query' => [
+                        'uenc' => $defaultStoreViewCode,
+                        '___from_store' => $this->formStoreMock,
+                        '___store' => $storeCode
+                    ]
+                ]
+            )
+            ->willReturnSelf();
+
+        $this->assertEquals(null, $this->redirectController->execute());
+    }
+
+    /**
+     *  Verify execute with exception
+     *
+     * @param string $defaultStoreViewCode
+     * @param string $storeCode
+     * @return void
+     * @dataProvider getConfigDataProvider
+     * @throws NoSuchEntityException
+     */
+    public function testRedirectWithThrowsException(string $defaultStoreViewCode, string $storeCode): void
+    {
+        $this->requestMock
+            ->expects($this->at(0))
+            ->method('getParam')
+            ->with(StoreResolver::PARAM_NAME)
+            ->willReturn($storeCode);
+        $this->requestMock
+            ->expects($this->at(1))
+            ->method('getParam')
+            ->with('___from_store')
+            ->willReturn($defaultStoreViewCode);
+        $this->storeRepositoryMock
+            ->expects($this->once())
+            ->method('get')
+            ->with($defaultStoreViewCode)
+            ->willThrowException(new NoSuchEntityException());
+        $this->messageManagerMock
+            ->expects($this->once())
+            ->method('addErrorMessage')
```

Comment:
@engcom-Echo 
I think purpose of covering this case with unit test - compare if we have correct message, I can't find where is it checked. Could you point me or adjust the test?

Reply:
Thank you.